### PR TITLE
DM USB: clean up the assert usage

### PR DESCRIPTION
**Pull request by xiaoguangwu**
Remove assert in USB mediator code.

Tracked-On: #3252
Signed-off-by: Xiaoguang Wu <xiaoguang.wu@intel.com>
Reviewed-by: Shuo A Liu <shuo.a.liu@intel.com>

### CODE REVIEW

**Comment by acrnsi** [comment]
OK to verify

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: devicemodel/hw/pci/xhci.c [hunk 1]
```diff
@@ -2024,7 +1984,6 @@ pci_xhci_cmd_address_device(struct pci_xhci_vdev *xdev,
 		dev_ctx->ctx_slot.dwSctx2, dev_ctx->ctx_slot.dwSctx3);
 
 	dev = XHCI_SLOTDEV_PTR(xdev, slot);
-	assert(dev != NULL);
```
Comment (dongyaozu) [inline devicemodel/hw/pci/xhci.c]: Will it be NULL?

Comment (xiaoguangwu) [reply devicemodel/hw/pci/xhci.c]: There are codes to process the NULL case before this assert.

1943         dev = xdev->slots[slot];
1944         if (!dev) {

**Comment by xiaoguangwu** [comment]
@dongyaozu  OK, in fact, this patch was reviewed and +1 in the mail list, I just forgot add the 'reviewed by' info, sorry for that and I will change the commit message. thanks for your reminding!

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Pull request merged**